providers.tf

```diff
@@ -1,3 +1,13 @@
 provider "aws" {
   region = "us-east-1"
+}
+
+data "aws_ecr_authorization_token" "token" {}
```

Comment:
This change introduces the `docker` provider, and configures it's access from what is locally available.
The [pipeline](https://github.com/IkenoXamos/aws-lab-infra/blob/main/modules/codepipeline/templates/buildspec_build.yml#L8) is already calling `aws ecr get-login-password`, which connects to AWS in order to access ECR.

This configuration simply leverages that. This is the recommended process as you can find [here](https://github.com/kreuzwerker/terraform-provider-docker/issues/3).